python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py

```diff
@@ -220,6 +223,8 @@ def test_execute_intervals():
     assert TestStepHandler.terminate_step_count == 0
     # every step should get checked at least once
     assert TestStepHandler.check_step_health_count >= 3
```

Comment:
@jmsanders @johannkm i'm trying to remove the protobuf pin from `dagster[test]` but this assertion fails when i upgrade it to 3.20.3 ([bk log](https://buildkite.com/dagster/dagster/builds/47456#01860a41-3aa5-417c-bb26-6d5a3b78689a/6-34)). 

i can't repro this in my local, which passes all the tests. git blame led me to https://dagster.phacility.com/D8218 where you two discussed about this assertion. do either of you know of why that happens and if we could modify the test so we drop the pin?